prow/job/sippy.py

```diff
@@ -228,3 +235,71 @@ def last2weeks(self):
     def last4weeks(self):
         last4weeks = self._now - timedelta(days=28)
         return last4weeks.strftime(self.starttime_format)
+
+
+class DataAnalyzer():
+
+    def __init__(self, payload: dict):
+        if payload:
+            self._data_set = payload
+        else:
+            raise ValueError("payload is empty")
+
+    def is_component_readiness_status_green(self):
+        '''
+          Check if any component has attribute `regressed_tests` and status >= -1
+        '''
+        logger.info('-' * 50)
+        logger.info("Start to analyze component readiness status")
+
+        issued_comps = []
+        rows = self._data_set.get("rows")
+        if len(rows):
+            for comp in rows:
+                name = comp.get("component")
+                cols = comp.get("columns")
+                for col in cols:
+                    status = col.get("status")
+                    if "regressed_tests" in col:
+                        regressed_tests = col.get("regressed_tests")
+                        logger.warning(
+                            f"{name} has regressed tests:\n{regressed_tests}\n")
+                        issued_comps.append(name)
+
+        if not issued_comps:
+            logger.info("No issued component found")
+
+        logger.info("Component readiness status is analyzed")
+
+        return len(issued_comps) == 0
+
+    def is_variants_status_green(self, expected_variants: [] = None):
```

Comment:
@rioliu-rh 

for api `api/variants`, no need to calculate it and use net_improvement directly in response.